lib/src/yaml_edit/src/errors.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Error thrown when a function is passed an invalid path.
+class PathError extends ArgumentError {
+  /// The path that caused the error
+  final Iterable<Object> path;
+
+  /// The last element of [path] that could be traversed.
+  Object parentNode;
+
+  PathError(this.path, Object invalidKeyOrIndex, this.parentNode,
```

Comment:
Should also so include the index of the parent in path?

If I pass a path `['a', 'a', 'a']`, then there is no way to known which key was actually problematic...